### PR TITLE
Fix memoryleak in io.fits.compressionmodule.

### PR DESCRIPTION
**Pull request by MSeifert04**
Fixes: #5399 

Related: 
- https://github.com/spacetelescope/PyFITS/issues/110

References: 
- https://docs.scipy.org/doc/numpy/user/c-info.how-to-extend.html#c.PyArray_SimpleNewFromData
- http://stackoverflow.com/questions/27912483/memory-leak-in-python-extension-when-array-is-created-with-pyarray-simplenewfrom

Note:
- The change cannot be backported to LTS because it requires numpy >= 1.7 and LTS still supports 1.6.

I checked the references and it seems the suggested fix (https://github.com/spacetelescope/PyFITS/issues/110#issuecomment-251710362) really does fix the issue, so I went ahead and implemented it.

Does anyone has any idea how to test for a non-python-object memoryleak in the tests?

### CODE REVIEW

**Comment by pllim** [comment]
@MSeifert04 , are you also planning to submit a fix to PyFITS repo? If not, stating that this fixes the PyFITS portion of the bug is misleading.

**Comment by pllim** [comment]
Definitely needs a change log. Probably hard to include a test. Not sure how far back to milestone this to.

**Comment by MSeifert04** [comment]
@pllim I'll add a changelog if I know the version for the entry 😄 

You're right about the PyFITS issue, I changed the wording.

**Comment by MSeifert04** [comment]
NumPy 1.6 support was dropped in 1.2: https://github.com/astropy/astropy/blob/master/CHANGES.rst#12-2016-06-19

**Comment by pllim** [comment]
So 1.2.2?

**Comment by MSeifert04** [comment]
There is probably some way to utilizie `NPY_VERSION`, `NPY_ABI_VERSION`, `NPY_FEATURE_VERSION`, `PyArray_GetNDArrayCVersion` or `PyArray_GetNDArrayCFeatureVersion` to preprocess or make a runtime-check for the additional line. I'll have a look 😄

**Comment by MSeifert04** [comment]
It's not as easy using `#if PY_MAJOR_VERSION == 2` in the source (or I'm doing something really stupid) and I haven't found any code in the wild that uses the numpy-version as preprocessor statement. 

If someone has any ideas I would be happy to change the code!

Fo now I'll leave it as is and I included a changelog entry for 1.2.2.

File: CHANGES.rst [hunk 1]
```diff
@@ -345,6 +345,8 @@ Bug Fixes
     float32 instead of float64 (with the E and F formats). These values are now
     always read as float64. [#5362]
 
+  - Fixed memoryleak when using the compression module. [#5399, 5464]
```
Comment (pllim) [inline CHANGES.rst]: The second PR is missing a `#`. I think there is a script somewhere that depends on it to autolink to issue. Maybe @bsipocz can confirm.

Comment (MSeifert04) [inline CHANGES.rst]: oups ...

Comment (bsipocz) [inline CHANGES.rst]: I haven't looked into the code, but I think our favourite astrobot checks the issue numbers and needs the ``#``

Comment (MSeifert04) [inline CHANGES.rst]: thanks for the confirmation. 

I've included the (forgotten) `#`.

**Comment by pllim** [comment]
Looks good to me but I'll leave it open for now if anyone else want to review. Please ping again before feature freeze if it is not merged soon enough. Thanks!

**Comment by MSeifert04** [comment]
> Please ping again before feature freeze if it is not merged soon enough.

oh, no. Not one of those again...

Is 1.2.2 released simultaneously with 1.3?

**Comment by pllim** [comment]
I can't answer that one. Maybe @astrofrog or @eteq can...

**Comment by astrofrog** [comment]
@MSeifert04 - not likely - if 1.2.2 happens it will be before 1.3. If 1.3 is released, the 1.2.x branch is closed so 1.2.2 wouldn't be released.

**Comment by pllim** [comment]
Fine, fine. I'll just merge this in now, so @MSeifert04 can sleep well tonight.

**Comment by MSeifert04** [comment]
@astrofrog Thanks for the clarification!

@pllim Thanks for merging.

**Pull request merged**